src/synapse-plugins/calculator-plugin/calculator-plugin-backend.vala

```diff
@@ -115,5 +101,42 @@ namespace Synapse {
                 return double.parse (solution);
             }
         }
+
+
+        private bool allowed_expression (string expr, out bool is_base_expr) {
+            is_base_expr = false;
+            if (expr.length > 6 &&
+                (expr.has_prefix ("ibase=") || expr.has_prefix ("obase="))
+            ) {
+                var suffix = expr.slice (6, expr.length);
+                var valid = int.parse (suffix) > 1 && int.parse (suffix) <= 16;
+                if (valid && expr.has_prefix ("ibase=")) {
+                    use_num = VALID_NUM.slice (0, int.parse (suffix));
+                }
+
+                is_base_expr = true;
+                return valid;
+            }
+
+            if (expr.length > 6 &&
+                (expr.has_prefix ("scale="))
+            ) {
+                var suffix = expr.slice (6, expr.length);
+                var valid = int.parse (suffix) >= 0;
+                is_base_expr = true;
+                return valid;
+            }
```

Comment:
There is a lot of similarities between the ibase, obase and scale logic. Maybe you can check has_prefix for all three, and then just adjust the suffix length check?

Reply:
I have DRY'd this now